base/shell/cmd/dir.c

```diff
@@ -973,14 +976,13 @@ DirPrintWideList(PDIRFINDINFO ptrFiles[],       /* [IN] Files' Info */
         if (ptrFiles[i]->stFindInfo.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY)
         {
             /* Directories need 2 additional characters for brackets */
-            if ((_tcslen(ptrFiles[i]->stFindInfo.cFileName) + 2) > iLongestName)
-                iLongestName = _tcslen(ptrFiles[i]->stFindInfo.cFileName) + 2;
+            cxWidth = ConGetTextWidth(ptrFiles[i]->stFindInfo.cFileName) + 2;
         }
         else
         {
-            if (_tcslen(ptrFiles[i]->stFindInfo.cFileName) > iLongestName)
-                iLongestName = _tcslen(ptrFiles[i]->stFindInfo.cFileName);
+            cxWidth = ConGetTextWidth(ptrFiles[i]->stFindInfo.cFileName);
         }
+        iLongestName = max(iLongestName, cxWidth);
```

Comment:
I've just realized that this (new) whole if-else block can be simplified further as follows:
```suggestion
        cxWidth = ConGetTextWidth(ptrFiles[i]->stFindInfo.cFileName);

        /* Directories need 2 additional characters for brackets */
        if (ptrFiles[i]->stFindInfo.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY)
            cxWidth += 2;

        iLongestName = max(iLongestName, cxWidth);
```

Reply:
Applying suggestions on deleted lines is not supported.

Reply:
Yes you'll have to add the suggestion manually by editing the file :stuck_out_tongue_closed_eyes:

Reply:
Done.